lib/konbata/models/upload_course.rb

```diff
@@ -204,52 +240,126 @@ def change_tab_visibility(tab, visible, tab_url_base)
     end
 
     ##
-    # Assembles the launch url with the course_id
+    # Returns the default whitelisted labels based on course type.
     ##
-    def scorm_launch_url(package_id)
-      "#{Konbata.configuration.scorm_launch_url}?course_id=#{package_id}"
+    def _default_whitelisted_labels
+      labels = [/home/i, /scorm/i]
+
+      if @type == :interactive
+        labels << /assignments/i
+      elsif @type == :non_interactive
+        labels << /modules/i
+      end
+
+      labels
     end
 
     ##
-    # Creates a scorm assignment using the Canvas api
+    # Queuries the course's content migration status and sleeps until the status
+    # says complete.
     ##
-    def create_scorm_assignment_external(upload_response, course_id)
-      url = scorm_launch_url(upload_response["package_id"])
+    def _wait_for_complete_migration
+      workflow_state = nil
 
-      payload = {
-        assignment__submission_types__: ["external_tool"],
-        assignment__integration_id__: upload_response["package_id"],
-        assignment__integration_data__: {
-          provider: "atomic-scorm",
-        },
-        assignment__external_tool_tag_attributes__: {
-          url: url,
-        },
-        assignment__points_possible__: POINTS_POSSIBLE,
-      }
+      until workflow_state =~ /complete/i
+        puts "Migration is #{workflow_state}..."
+        sleep(2)
```

Comment:
This is kinda kludgy, and would probably be better in a asynchronous Sidekiq/Resque/Delayed_Job workflow.  Given time/resource constraints, this is fine for now.  I will list this as a "bug" to be fixed when we add the additional formats to handle.